### DISCUSSION TITLE
Source question: after zoomIOut, don't you need to call Setviewport?

### DISCUSSION THREAD

**Discussion by vaynevayne** [xyflow/xyflow#4514] [Q&A]
hello, Please advise

https://github.com/xyflow/xyflow/blob/69f53c0eff5962498c68f63232bf8f1076255b85/packages/react/src/hooks/useViewportHelper.ts#L35

**Comment by bcakmakoglu** [comment]
Advise on what though?

Are you experiencing some issue? If yes, please describe the issue in detail and provide a reproduction.
Are you trying to implement something but have trouble doing so? If yes, please describe what you're trying to build in detail and also describe what you've tried and what hasn't worked.

**Reply by vaynevayne** [reply]
I am learning from xy-flow to implement react-fabricjs,
Can you explain why this code does not require setViewport?

**Reply by bcakmakoglu** [reply]
`setViewport` -> used to set `x`, `y` *and* `zoom`.
`zoomTo` (uses `.scaleTo`) -> used to only set `zoom` level
